main/tasks.py

```diff
@@ -273,3 +277,102 @@ def email_monthly_charges_report() -> None:
+        print("Clockify API key not found in environment variables")
+        return
+
+    days_back = 30
+    api = ClockifyAPI(api_key)
+    end_date = timezone.now()
+    start_date = end_date - datetime.timedelta(days=days_back)
+
+    project_ids = list(
+        Project.objects.exclude(clockify_id="").values_list("clockify_id", flat=True)
+    )
+
+    total_entries_synced = 0
+    total_entries_skipped = 0
+
+    for project_id in project_ids:
+        try:
+            print(f"Processing project ID: {project_id}")
+            payload = {
+                "dateRangeStart": start_date.strftime("%Y-%m-%dT00:00:00.000Z"),
+                "dateRangeEnd": end_date.strftime("%Y-%m-%dT23:59:59.000Z"),
+                "detailedFilter": {"page": 1, "pageSize": 200},
+                "projects": {"contains": "CONTAINS", "ids": [project_id]},
+            }
+
+            response = api.get_time_entries(payload)
+            entries = response.get("timeentries", [])
+            if not isinstance(entries, list):
+                entries = []
+
+            project_entries_synced = 0
+            project_entries_skipped = 0
+
+            for entry in entries:
+                entry_id = entry.get("id") or entry.get("_id")
+                project_id = entry.get("projectId")
+                user_email = entry.get("userEmail")
+                time_interval = entry.get("timeInterval", {})
+                start = time_interval.get("start")
+                end = time_interval.get("end")
+
+                if not (project_id and user_email and start and end):
+                    continue
+
+                try:
+                    project = Project.objects.get(clockify_id=project_id)
+                except Project.DoesNotExist:
+                    print(f"Project with clockify_id {project_id} not found.")
+                    continue
+
+                try:
+                    user = User.objects.get(email=user_email)
+                except User.DoesNotExist:
+                    print(f"User with email {user_email} not found.")
+                    continue
+
+                start_time = datetime.datetime.fromisoformat(
+                    start.replace("Z", "+00:00")
+                )
+                end_time = datetime.datetime.fromisoformat(end.replace("Z", "+00:00"))
+
+                existing_entry = TimeEntry.objects.filter(
+                    user=user,
+                    project=project,
+                    start_time=start_time,
+                    end_time=end_time,
+                ).exists()
+
+                if existing_entry:
+                    project_entries_skipped += 1
+                    continue
+
+                TimeEntry.objects.create(
+                    user=user,
+                    project=project,
+                    start_time=start_time,
+                    end_time=end_time,
+                    clockify_id=entry_id,
+                )
+                project_entries_synced += 1
```

Comment:
This can be done simpler:
```suggestion
                TimeEntry.objects.get_or_create(
                    clockify_id=entry_id,
                    defaults={
                        "user"=user,
                        "project"=project,
                        "start_time"=start_time,
                        "end_time"=end_time,
                    }
                )
```
We don't really care if it has been created or if it existed, as long as it is not duplicated. Check [`get_or_create` docs](https://docs.djangoproject.com/en/5.2/ref/models/querysets/#get-or-create).